bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -1110,6 +1110,17 @@ void PreProcessor::cancelPreProcessing(NodeIdType clientId, uint16_t reqOffsetIn
   }
 }
 
+PreProcessResultSignature signPrimaryPreExecutionResultHash(const ReplicaId &myId,
```

Comment:
A native place for this function is RequestProcessingState, where we deal with hashes and signatures.

Reply:
Good point! I felt this function is out of place but I wanted to defer the signature calculation. However considering that 99% of the runs will be happy cases - my optimisation makes no sense.

Please check my second commit. It implements your suggestion.